tests/e2e/common/common.go

```diff
@@ -168,50 +168,56 @@ func DefaultOffChainConfigParamsFromNodes(nodes []client.Chainlink) (contracts.O
 	}, nkb, nil
 }
 
-func CreateBridges(nodes []client.Chainlink, mock *client.MockserverClient) ([]BridgeInfo, error) {
+func CreateBridges(contracts []string, nodes []client.Chainlink, mock *client.MockserverClient) (map[string][]BridgeInfo, error) {
 	relayConfig := map[string]string{
 		"nodeType":      "terra",
 		"tendermintURL": "http://terrad:26657",
 		"fcdURL":        "http://fcd-api:3060",
 		"chainID":       "localterra",
 	}
-	bi := make([]BridgeInfo, 0)
-	for nIdx, n := range nodes {
-		sourceValueBridge := client.BridgeTypeAttributes{
-			Name:        "variable",
-			URL:         fmt.Sprintf("%s/node%d", mock.Config.ClusterURL, nIdx),
-			RequestData: "{}",
-		}
-		observationSource := client.ObservationSourceSpecBridge(sourceValueBridge)
-		err := n.CreateBridge(&sourceValueBridge)
-		if err != nil {
-			return nil, err
-		}
-		juelsBridge := client.BridgeTypeAttributes{
-			Name:        "juels",
-			URL:         fmt.Sprintf("%s/juels", mock.Config.ClusterURL),
-			RequestData: "{}",
+	biMap := make(map[string][]BridgeInfo)
+	for _, contract := range contracts {
+		for _, n := range nodes {
+			nodeContractPairID, err := BuildNodeContractPairID(n, contract)
+			if err != nil {
+				return nil, err
+			}
+			sourceValueBridge := client.BridgeTypeAttributes{
+				Name:        nodeContractPairID,
+				URL:         fmt.Sprintf("%s/%s", mock.Config.ClusterURL, nodeContractPairID),
+				RequestData: "{}",
+			}
+			observationSource := client.ObservationSourceSpecBridge(sourceValueBridge)
+			err = n.CreateBridge(&sourceValueBridge)
+			if err != nil {
+				return nil, err
+			}
+			juelsBridge := client.BridgeTypeAttributes{
+				Name:        "juels",
+				URL:         fmt.Sprintf("%s/juels", mock.Config.ClusterURL),
+				RequestData: "{}",
+			}
+			juelsSource := client.ObservationSourceSpecBridge(juelsBridge)
+			err = n.CreateBridge(&juelsBridge)
+			if err != nil {
+				return nil, err
+			}
+			_, err = n.CreateTerraChain(&client.TerraChainAttributes{ChainID: "localterra"})
+			if err != nil {
+				return nil, err
+			}
+			if _, err = n.CreateTerraNode(&client.TerraNodeAttributes{
+				Name:          "terra",
+				TerraChainID:  relayConfig["chainID"],
+				TendermintURL: relayConfig["tendermintURL"],
+				FCDURL:        relayConfig["fcdURL"],
```

Comment:
`FCDURL` is not part of the node's config (represented by `TerraNodeAttributes`) anymore (for a while now), but actually a part of chains config (represented by `TerraChainAttributes`).

Reply:
Here is the relevant issue @gheorghestrimtu:

-  https://github.com/smartcontractkit/chainlink-terra/issues/78

Reply:
It would be great if you can move this along as part of this effort 🙌

Reply:
Of course, no problem